### DISCUSSION TITLE
[RESOLVED] Can't run codeql outside the starter project

### DISCUSSION THREAD

**Discussion by CaledoniaProject** [github/codeql#8494] [General]
I'm trying to avoid the starter project by creating an empty one.

I first copy the qlpack.yaml out:

```
name: getting-started/codeql-extra-queries-cpp
version: 0.0.0
dependencies:
  codeql/cpp-all: "*"
```

Then create a simple ql file and run it against a database, and codeql couldn't find the built-in package:

```
ERROR: Referenced pack 'codeql/cpp-all' not found
```

What additional steps is required to make it work?

**Comment by smowton** [comment]
Have you tried https://codeql.github.com/docs/codeql-for-visual-studio-code/setting-up-codeql-in-visual-studio-code/#updating-an-existing-workspace-for-codeql ?

**Reply by CaledoniaProject** [reply]
If I understood correctly, codeql libraries like cpp-all are not shipped with codeql binaries releases?

**Comment by CaledoniaProject** [comment]
Ah I just figured out vscode-codeql-starter actually contains a git submodule of codeql repo. Case closed.
